doc/HLD_Version_Numbers.rst

```diff
@@ -0,0 +1,231 @@
+
+- [r.verno.dtm]: this is a consequence of [r.verno.serial] fulfillment: if updates are replayed in the order of their version numbers, original serial history is exactly reproduced; 
+
+- [r.verno.optimistic-concurrency]: version numbers can be used for scalable generation of pseudo-times are discussed in [1]; 
+
+- [r.verno.update-streams]: as described in the cache re-integration section of the Functional specification, update stream mechanism can be implementing by introducing a special unit tracking re-integration state. This unit servers as a transactional stream caret, tracking re-integration progress in the presence of failures; 
+
+- [r.verno.fol]: by design, version number contains lsn field.
+
+Dependencies
+===============
+
+- [r.dtx]: distributed transactions are supported 
+
+  - [r.dtx.units]: file system updates acts on units 
+
+  - [r.dtx.units.identity]: units affected by a given update are identifiable 
+
+  - [R.DTX.SERIALIZABLE]: updates are serializable 
+
+- [r.resource]: scalable distributed resource management is implemented 
+
+  - [r.resource.cacheable]: resources are cacheable by consumers 
+
+- rpc: 
+
+  - [R.NET.RPC.STREAMS.MULTIPLE]: multiple update concurrent streams between a given pair of nodes are supported 
+
+- fol 
+
+  - [r.fol.lsn]: fol records are identifiable by lsn 
+
+  - [r.fol.verno]: fol record stores before- and after- version numbers 
+
+  - [r.fol.lsn.compare]: lsn numbers can be compared 
+
+- back-end 
+
+  - [r.back-end.store-verno]: back end stores current version number as object attribute 
+
+  - [R.BACK-END.TRANSACTIONAL]: back-end supports local transactions
+  
+Security Model
+===============
+
+A node can wreak chaos in the system by spoofing version numbers, so they must be protected by the same means as the rest of meta-data.
+
+Refinement
+===============
+
+Version number component has little internal logic. 
+
+***************
+State
+***************
+
+See version number comparison invariant above. 
+
+Use Cases (Scenarios)
+=====================
+
+Scenario 1
+
+.. image:: Images/scev1.PNG
+
+Scenario 2
+
+.. image:: Images/scev2.PNG
+
+Scenario 3
+
+.. image:: Images/scev3.PNG
+
+Scenario 4
+
+.. image:: Images/scev4.PNG
+
+
+***************
+Failures
+***************
+
+This specification is all about handling failures. 
+
+Analysis
+===============
+
+The hybrid operation mode introduced above, disentangles issues of resource management and concurrency control from failure recovery, which traditional session-slot based protocols fail to do. It's also instructive to compare version maintenance with the epoch algorithm: version numbers are, roughly speaking, per-object epochs.
+
+
+
```

Comment:
The paragraphs and titles, are organized in a properly level.
They are all correct.
Please follow this one to modify PR #193, PR #196 and PR #201

Reply:
Sure